### ISSUE TITLE
auxk loss: sign flip for correct prediction of the residual

### ISSUE THREAD

**Issue by fel-thomas** [EleutherAI/sparsify#48]
# Fix auxiliary loss sign for correct residual prediction 

Hi, thx for all your open source projects :) 

I think the current loss is pushing the auxilary towards 'not predicting the residual' and we need to invert the way the `e` variable is defined. 

## Issue

The current implementation defines the residual as:

```python
e = sae_out - y
auxk_loss = sum((e_hat - e)**2)
# if we expand sum(e_hat - sae_out + y)**2
```

This effectively minimizes `(e_hat - sae_out + y)^2`, which does not correctly drive the auxiliary prediction `e_hat` to be the needed correction.

## Desired Behavior

We want the correction `e_hat` to satisfy:

```python
y_pred + e_hat = y   # i.e., e_hat should be y - y_pred
# y_pred is 'sae_out'
```

## Proposed Change

Change the residual definition to:

```python
e = y - y_pred
auxk_loss = sum((e_hat - (y - y_pred))**2)
```
as right now the implementation **is exactly the negative of the target.**

Now, minimizing the new loss will directly push `e_hat` towards `y - y_pred`.

This update aligns the implementation with the intended formulation and the description in the OpenAI paper.

I opened a pr here: https://github.com/EleutherAI/sparsify/pull/47.

Congrats again on the libs.

**Comment by norabelrose** [comment]
Fixed by #47
